### DISCUSSION TITLE
Wagmi connector support Binance Wallet?

### DISCUSSION THREAD

**Discussion by ChrisChan888** [wevm/wagmi#2023] [Q&A]
Will wagmi connector support Binance Wallet?

**Comment by jeftarmascarenhas** [comment]
Hi @ChrisChan888 see this code below that create a custom `injectedConnector` to Binance.
https://gist.github.com/jeftarmascarenhas/3214feb7c6871253cf9486fd8ccacc40

**Reply by artginzburg** [reply]
This doesn't seem to work though. At first, this.bsc was always undefined. I made sure that it gets `{ bsc } from 'wagmi/chains'`.

Now I get two errors in the console: 
```
TypeError: provider.request is not a function. (In 'provider.request({
      method: "eth_accounts"
    })', 'provider.request' is undefined)
```
and
```
Unhandled Promise Rejection: TypeError: Attempting to change the setter of an unconfigurable property.
```
Idk why the first one occurs, but the second one seems to be about `Object.defineProperty` in your code

**Reply by tmm** [reply]
Not familiar with Binance Wallet, but might be able to use `InjectedConnector`'s `getProvider` property.

```ts
import { InjectedConnector } from 'wagmi/connectors/injected'
 
const connector = new InjectedConnector({
  options: {
    name: 'Binance Wallet',
    getProvider: () =>
      typeof window !== 'undefined' ? window.BinanceChain : undefined,
  },
})
```

**Reply by artginzburg** [reply]
What is `window.BinanceChain`? I'm trying to get that working in React. I don't even have `window.ethereum`

**Reply by tmm** [reply]
> What is `window.BinanceChain`?

Based on [this earlier comment](https://github.com/wagmi-dev/wagmi/discussions/2023#discussioncomment-5472909), it looks like Binance Wallet injects it's provider at `window.BinanceChain`. Again, I'm not familiar with Binance Wallet so just a guess.

**Reply by 0xjojoex** [reply]
I believed the above gist was copied from pancakeswap https://github.com/pancakeswap/pancake-frontend/blob/develop/packages/wagmi/connectors/binanceWallet/binanceWallet.ts
When getProvider wasn't an option yet at that time. 
I can open a PR if it's needed

**Reply by artginzburg** [reply]
Wow, that means we can `import { BinanceWalletConnector } from '@pancakeswap/wagmi/connectors/binanceWallet'`. Good to know. Thank you very much!
